optuna/cli.py

```diff
@@ -51,6 +56,27 @@ def _check_storage_url(storage_url: Optional[str]) -> str:
     raise CLIUsageError("Storage URL is not specified.")
 
 
+def _get_storage(storage_url: Optional[str], storage_class: Optional[str]) -> BaseStorage:
+    storage_url = _check_storage_url(storage_url)
+    if storage_class:
+        if storage_class == JournalRedisStorage.__name__:
+            return JournalStorage(JournalRedisStorage(storage_url))
+        if storage_class == JournalFileStorage.__name__:
+            return JournalStorage(JournalFileStorage(storage_url))
+        if storage_class == RDBStorage.__name__:
+            return RDBStorage(storage_url)
+        raise CLIUsageError("Unsupported storage class")
+
+    if storage_url.startswith("redis"):
+        return JournalStorage(JournalRedisStorage(storage_url))
+    if os.path.isfile(storage_url):
+        return JournalStorage(JournalFileStorage(storage_url))
```

Comment:
This implementation expects an existing file, so the following command fails if `journal.log` does not exist.

```console
$ optuna create-study --storage ./journal.log
[E 2023-06-13 14:10:56,836] Failed to guess storage class from storage_url
```

How about accepting newly created files?